packages/shared/src/hooks/useReadingRank.ts

```diff
@@ -77,12 +85,11 @@ export default function useReadingRank(): ReturnType {
       neverShowRankModal: newNeverShowRankModal,
     });
 
-  const shouldShowRankModal =
-    !neverShowRankModal ??
-    checkShouldShowRankModal(
-      alerts.rankLastSeen,
-      cachedRank?.rank?.lastReadTime || remoteRank?.rank?.lastReadTime,
-    );
+  const shouldShowRankModal = checkShouldShowRankModal(
+    alerts.rankLastSeen,
+    cachedRank?.rank?.lastReadTime || remoteRank?.rank?.lastReadTime,
+    neverShowRankModal,
+  );
```

Comment:
Yes, that is very important. Though in this part, the `null` or `undefined` checks are already in place, and the function that decides also has the conditions if the parameter is `null` or `undefined` for both parameters.

For the case of `alerts` object, we set default values on it so while it is still running the async, it will have always have the placeholder value for `alerts.rankLastSeen`.

The same goes for `neverShowRankModal`, it was defined with optional chaining which `null` or `undefined` condition was checked in the function as well.

I thought it was better to do the decision in the function itself, so we won't do it in the consumer level of this hook.

Though it also makes sense to do this at the consumer level (if `undefined`) it is just I thought of it to be much easier this way for consumers/devs.

What do you think?